index.js

```diff
@@ -492,21 +492,42 @@ async function sendInitialChatbotMessage(session) {
     locationData = await db.getLocationData(location)
     await sendTwilioMessage(locationData.twilio_number, session.phonenumber, `Please check on the bathroom. Please respond with 'ok' once you have checked on it.`);
     await db.startChatbotSessionState(session);
-    setTimeout(reminderMessage, locationData.unrespondedTimer, session.locationid);
+    setTimeout(reminderMessage, locationData.unrespondedTimer, session.sessionid);
+    setTimeout(fallbackMessage, locationData.unresponded_Session_Timer, session.sessionid)
 }
 
-async function reminderMessage(location) {
-    console.log("Reminder message being sent");
-    let session = await db.getMostRecentSession(location); // Gets the updated state for the chatbot
-    locationData = await db.getLocationData(location)
-    if(session.chatbot_state == 'Started') {
-        //send the message
-        await sendTwilioMessage(locationData.twilio_number, session.phonenumber, `This is a reminder to check on the bathroom`)
-        session.chatbot_state = 'Waiting for Response';
-        await db.saveChatbotSession(session);
-    }
-    //else do nothing
+async function reminderMessage(sessionid) {
+  console.log("Reminder message being sent");
+  let session = await db.getSessionWithSessionId(sessionid); // Gets the updated state for the chatbot
+  var location = session.locationid;
+  locationData = await db.getLocationData(location)
```

Comment:
I would move this DB call into your the body of your if statement. Database calls are expensive in terms of time. We want to avoid making more calls than necessary. In this case, we don't need to know the locationData unless the chatbot_state indicates that we need to send the message, right?

Reply:
Done for both functions... And I'll keep the cost of function calls in mind going forward :)